app/controllers/source_data_controller.rb

```diff
@@ -86,4 +95,14 @@ def show
     @klass.joins(:data_source).
       merge(GrdaWarehouse::DataSource.editable_by(current_user).source)
   end
+
+  private def hmis_url
+    return unless @data_source.hmis
+
+    base = "https://#{@data_source.hmis}"
+    return "#{base}/projects/#{@item.id}" if @item.is_a?(GrdaWarehouse::Hud::Project)
+    return "#{base}/organizations/#{@item.id}" if @item.is_a?(GrdaWarehouse::Hud::Organization)
+    return "#{base}/client/#{@item.id}" if @item.is_a?(GrdaWarehouse::Hud::Client)
+    return "#{base}/client/#{@item.client.id}/enrollments/#{@item.enrollment.id}" if @item.respond_to?(:enrollment) && @item.respond_to?(:client)
+  end
```

Comment:
Also, I think it's bad to use `_url` since that will collide with built-in helpers.
```suggestion
  private def hmis_url_for(item)
    return unless item&.data_source&.hmis?

    base = "https://#{item.data_source.hmis}"
    return "#{base}/projects/#{item.id}" if item.is_a?(GrdaWarehouse::Hud::Project)
    return "#{base}/organizations/#{item.id}" if item.is_a?(GrdaWarehouse::Hud::Organization)
    return "#{base}/client/#{item.id}" if item.is_a?(GrdaWarehouse::Hud::Client)
    return "#{base}/client/#{item.client.id}/enrollments/#{item.enrollment.id}" if item.respond_to?(:enrollment) && item.respond_to?(:client)
  end
```

Reply:
Where do you think this function should go? It feels kind of random to have it in this specific controller. like you said, we may want to show these elsewhere. I just don't know what the appropriate class is for it.

Reply:
maybe it can be on `GrdaWarehouse::Hud::Base`?

Reply:
Or on the DataSource model

Reply:
Yeah, for now I think `GrdaWarehouse::DataSource` makes the most sense, either that or some sort of helper class specific to the HMIS.